### ISSUE TITLE
Incorrect and inconsistent ehaviour of list control

### ISSUE THREAD

**Issue by oneeyeman1** [wxWidgets/wxWidgets#23387]
### Description
Trying to select/unselect row in list control gives incorrect and inconsistent behavior

**Expected vs observed behavior:**
I expected all 3 platforms to behave consistently
But native MSW controls behaves differently than generic one on GTK/OSX

**Patch to the minimal sample::**
```
Igor@fordanwork MINGW64 /c/wxFork (GetCursorFromBundleResoure_part2)
$ diff -bru samples/minimal/minimal.cpp c:/wxWidgets/samples/minimal/minimal.cpp
--- samples/minimal/minimal.cpp 2022-12-31 22:01:42.599105800 -0600
+++ c:/wxWidgets/samples/minimal/minimal.cpp    2023-03-28 09:12:30.610335400 -0500
@@ -25,6 +25,7 @@
 #ifndef WX_PRECOMP
     #include "wx/wx.h"
 #endif
+#include "wx/listctrl.h"

 // ----------------------------------------------------------------------------
 // resources
@@ -50,7 +51,7 @@
     // this one is called on application startup and is a good place for the app
     // initialization (doing it here and not in the ctor allows to have an error
     // return: if OnInit() returns false, the application terminates)
-    virtual bool OnInit() override;
+    virtual bool OnInit() wxOVERRIDE;
 };

 // Define a new frame type: this is going to be our main frame
@@ -112,6 +113,42 @@
 // the application class
 // ----------------------------------------------------------------------------

+class MyDialog : public wxDialog
+{
+public:
+    MyDialog(wxWindow *parent) : wxDialog(parent, wxID_ANY, "Test")
+    {
+        list = new wxListCtrl( this, wxID_ANY, wxDefaultPosition, wxDefaultSize, wxLC_REPORT );
+        list->AppendColumn( "" );
+        list->InsertItem( 0, "Test 0" );
+        list->InsertItem( 1, "Test 1" );
+        list->InsertItem( 2, "Test 2" );
+        list->InsertItem( 3, "Test 3" );
+        list->Bind( wxEVT_LEFT_DOWN, &MyDialog::OnSelectDeselectField, this );
+    }
+    void OnSelectDeselectField(wxMouseEvent &event)
+    {
+        int flags;
+        auto item = list->HitTest( event.GetPosition(), flags );
+        if( item != wxNOT_FOUND )
+        {
+            auto label = list->GetItemText( item );
+            auto state = list->GetItemState( item, wxLIST_STATE_SELECTED );
+            if( state & wxLIST_STATE_SELECTED )
+            {
+                list->SetItemState( item, 0, wxLIST_STATE_SELECTED );
+                list->SetItemState( item, 0, wxLIST_STATE_FOCUSED );
+            }
+            else
+            {
+                list->SetItemState( item, wxLIST_STATE_FOCUSED, wxLIST_STATE_FOCUSED );
+                event.Skip();
+            }
+        }
+    }
+private:
+    wxListCtrl *list;
+};
 // 'Main program' equivalent: the program execution "starts" here
 bool MyApp::OnInit()
 {
@@ -188,6 +225,8 @@

 void MyFrame::OnAbout(wxCommandEvent& WXUNUSED(event))
 {
+    MyDialog dlg(this);
+    dlg.ShowModal();
     wxMessageBox(wxString::Format
                  (
                     "Welcome to %s!\n"

```
**To reproduce:**
Run the sample
Select Help->About
In the dialog click on the "Item 0", then "Item 1" then "Item 2"

On Windows only clicked item will be highlighted
On GTK/OSX clicking first item will highlight it, clicking second item will highlight it, keeping the highlight of the first one, But when third item is clicked - only third item wil have highlight.

### Platform and version information

- wxWidgets version  you use: 3.2.0
- wxWidgets port  you use: MSW, GTK and OSX
- OS  and its version: Win8.1, OSX 10.13, Gentoo Linux 

+ GTK version:  3.24.29
+ Which GDK  backend is used: Wayland
+ Current theme:

**Comment by taler21** [comment]
Maybe fixed by fc357a49ef162f8c023478a5472d1805dcf5e2ad (Fix missing deselection of single selected but not focused item in wxGenericListCtrl, 2024-02-25).

**Comment by vadz** [comment]
Yes, I think it's the same bug, thanks!
